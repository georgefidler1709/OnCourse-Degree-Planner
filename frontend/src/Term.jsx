import React, {Component} from 'react';
import styled from 'styled-components';
import { Droppable } from 'react-beautiful-dnd';
import Course from './Course';

const Container = styled.div`
  margin: 8px;
  border: 1px solid lightgrey;
  border-radius: 2px;
  width: 220px;

  display: flex;
  flex-direction: column;
`;
const Title = styled.h3`
  padding: 8px;
`;
const CourseList = styled.div`
  padding: 8px;
  flex-grow: 1;
  min-height: 100px;
`;

class Term extends Component {
  render() {
    return (
      <Container>
        <Title>{this.props.termId}</Title>
        <Droppable droppableId={this.props.termId}>
          {provided => (
            <CourseList innerRef={provided.innerRef} {...provided.droppableProps}>
              {this.props.courses.map((course, index) => {
                const courseTag = `${course.subject.toString()}${course.code.toString()}`
                return <Course key={courseTag} courseId={courseTag} course={course} index={index} />
              }
                
              )}
              {provided.placeholder}
            </CourseList>
          )}
        </Droppable>
      </Container>
    );
  }
}

export default Term
