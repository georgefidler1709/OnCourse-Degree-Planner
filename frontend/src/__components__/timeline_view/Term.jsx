import React, {Component} from 'react';
import styled from 'styled-components';
import { Droppable } from 'react-beautiful-dnd';
import Course from './Course';

const Container = styled.div`
  margin: 8px;
  border: 1px solid lightgrey;
  border-radius: 2px;
  width: 220px;

  display: flex;
  flex-direction: column;
`;
const Title = styled.h3`
  padding: 8px;
`;
const CourseList = styled.div`
  padding: 8px;
  flex-grow: 1;
  min-height: 100px;
`;

class Term extends Component {
  render() {
    return (
      <Container>
        <Title>{this.props.term_id}</Title>
        <Droppable droppableId={this.props.term_id}>
          {provided => (
            <CourseList innerRef={provided.innerRef} {...provided.droppableProps}>
              {this.props.courses.map((course, index) => {
                return <Course key={course.course_id} course_id={course.course_id} course={course.course_info} index={index} />
              }
                
              )}
              {provided.placeholder}
            </CourseList>
          )}
        </Droppable>
      </Container>
    );
  }
}

export default Term
