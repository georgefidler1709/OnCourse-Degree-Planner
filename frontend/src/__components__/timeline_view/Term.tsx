import React from 'react';
import styled from 'styled-components';
import { Droppable } from 'react-beautiful-dnd';
import Course from './Course';
import { Course as ApiCourse } from '../../Api';

const Container = styled.div`
  margin: 8px;
  border: 1px solid lightgrey;
  border-radius: 2px;
  width: 220px;

  display: flex;
  flex-direction: column;
`;
const Title = styled.h3`
  padding: 8px;
`;
const CourseList = styled.div`
  padding: 8px;
  flex-grow: 1;
  min-height: 100px;
`;

interface TermProps {
  key: string;
  courses: Array<ApiCourse>;
}

function Term(props: TermProps) {
  return (
    <Container>
      <Title>{props.key}</Title>
      <Droppable droppableId={props.key}>
        {provided => (
          <CourseList innerRef={provided.innerRef} {...provided.droppableProps}>
            {props.courses.map((course, index) => {
              let code = course.code.toString();
              return <Course 
                {...course}
                course_id={code}
                key={code} 
                index={index} />
            }
            )}
            {provided.placeholder}
          </CourseList>
        )}
      </Droppable>
    </Container>
  );
}

export default Term
