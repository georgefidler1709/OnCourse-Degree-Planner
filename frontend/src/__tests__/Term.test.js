import React from 'react'
import { shallow, mount } from 'enzyme';
import { DragDropContext } from 'react-beautiful-dnd';
import Term from '../__components__/timeline_view/Term';

console.error = jest.fn();
console.warn = jest.fn();

describe("Won't render correctly outside of a DragDropContext", () => {
    it('renders correctly', () => {
      const wrapper = shallow(<Term 
                                  key="test" 
                                  termId="2 2019" 
                                  courses={[]}
                                  highlight={false} 
                                  removeCourse={(s) => {return}}
                                  getError={(s) => undefined}
                                  getWarn={(s) => undefined}
                                />);
      expect(wrapper).toMatchSnapshot();
    });

    it('Will render correctly within a DropDropContext', () => {
      const wrapper = mount(
      <DragDropContext>
      {
        <Term key="test" termId="2 2019" courses={[]}/>
      }
      </DragDropContext>);
      expect(wrapper).toMatchSnapshot();
    });

    it('Shows correct term title', () => {
      const wrapper = mount(
      <DragDropContext>
      {
        <Term key="test" termId="2 2019" courses={[]}/>
      }
      </DragDropContext>);
      expect(wrapper.find(Term).first().text()).toBe('T2 2019')
    });
});