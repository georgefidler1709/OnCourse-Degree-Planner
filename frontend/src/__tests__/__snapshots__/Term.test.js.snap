// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Won't render correctly outside of a DragDropContext Will render correctly within a DropDropContext 1`] = `
<DragDropContext>
  <ErrorBoundary>
    <App
      contextId="0"
      liftInstruction="Draggable item. Ensure your screen reader is not in browse mode and then press space bar to lift."
      setCallbacks={[Function]}
    >
      <Provider
        context={
          Object {
            "$$typeof": Symbol(react.context),
            "Consumer": Object {
              "$$typeof": Symbol(react.context),
              "_calculateChangedBits": null,
              "_context": [Circular],
            },
            "Provider": Object {
              "$$typeof": Symbol(react.provider),
              "_context": [Circular],
            },
            "_calculateChangedBits": null,
            "_currentRenderer": Object {},
            "_currentRenderer2": null,
            "_currentValue": null,
            "_currentValue2": null,
            "_threadCount": 0,
          }
        }
        store={
          Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
        }
      >
        <Term
          courses={Array []}
          key="test"
          termId="2 2019"
        >
          <styled.div>
            <div
              className="sc-bwzfXH itFJcJ"
            >
              <styled.h3>
                <h3
                  className="sc-htpNat jumdkD"
                >
                  T2 2019
                </h3>
              </styled.h3>
              <ConnectFunction
                direction="vertical"
                droppableId="2 2019"
                getContainerForClone={[Function]}
                ignoreContainerClipping={false}
                isCombineEnabled={false}
                isDropDisabled={false}
                mode="standard"
                renderClone={null}
                type="DEFAULT"
              >
                <Droppable
                  direction="vertical"
                  droppableId="2 2019"
                  getContainerForClone={[Function]}
                  ignoreContainerClipping={false}
                  isCombineEnabled={false}
                  isDropDisabled={false}
                  mode="standard"
                  placeholder={null}
                  renderClone={null}
                  shouldAnimatePlaceholder={false}
                  snapshot={
                    Object {
                      "draggingFromThisWith": null,
                      "draggingOverWith": null,
                      "isDraggingOver": false,
                      "isUsingPlaceholder": false,
                    }
                  }
                  type="DEFAULT"
                  updateViewportMaxScroll={[Function]}
                  useClone={null}
                >
                  <styled.div
                    data-rbd-droppable-context-id="0"
                    data-rbd-droppable-id="2 2019"
                    innerRef={[Function]}
                    isDraggingOver={false}
                  >
                    <div
                      className="sc-bxivhb hYJMoY"
                      data-rbd-droppable-context-id="0"
                      data-rbd-droppable-id="2 2019"
                    >
                      <AnimateInOut
                        on={null}
                        shouldAnimate={false}
                      />
                    </div>
                  </styled.div>
                </Droppable>
              </ConnectFunction>
            </div>
          </styled.div>
        </Term>
      </Provider>
    </App>
  </ErrorBoundary>
</DragDropContext>
`;

exports[`Won't render correctly outside of a DragDropContext renders correctly 1`] = `
<styled.div>
  <styled.h3>
    T2 2019
  </styled.h3>
  <Connect(Droppable)
    direction="vertical"
    droppableId="2 2019"
    getContainerForClone={[Function]}
    ignoreContainerClipping={false}
    isCombineEnabled={false}
    isDropDisabled={false}
    mode="standard"
    renderClone={null}
    type="DEFAULT"
  >
    <Component />
  </Connect(Droppable)>
</styled.div>
`;
